src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaComparePublishChangesOperation.cs

```diff
@@ -1,80 +1,36 @@
+    abstract class SchemaComparePublishChangesOperation : ITaskOperation
     {
-        private CancellationTokenSource cancellation = new CancellationTokenSource();
-
-        /// <summary>
-        /// Gets the unique id associated with this instance.
-        /// </summary>
         public string OperationId { get; private set; }
 
-        public SchemaComparePublishChangesParams Parameters { get; }
-
-        protected CancellationToken CancellationToken { get { return this.cancellation.Token; } }
-
-        public string ErrorMessage { get; set; }
-
         public SqlTask SqlTask { get; set; }
 
         public SchemaComparisonResult ComparisonResult { get; set; }
 
-        public SchemaComparePublishResult PublishResult { get; set; }
+        public string ErrorMessage { get; set; }
 
-        public SchemaComparePublishChangesOperation(SchemaComparePublishChangesParams parameters, SchemaComparisonResult comparisonResult)
-        {
-            Validate.IsNotNull("parameters", parameters);
-            this.Parameters = parameters;
-            Validate.IsNotNull("comparisonResult", comparisonResult);
-            this.ComparisonResult = comparisonResult;
-        }
+        protected CancellationToken CancellationToken { get { return cancellation.Token; } }
 
-        public void Execute(TaskExecutionMode mode)
-        {
-            if (this.CancellationToken.IsCancellationRequested)
-            {
-                throw new OperationCanceledException(this.CancellationToken);
-            }
+        protected readonly CancellationTokenSource cancellation = new();
 
-            try
-            {
-                this.PublishResult = this.ComparisonResult.PublishChangesToTarget(this.CancellationToken);
-                if (!this.PublishResult.Success)
-                {
-                    // Sending only errors and warnings - because overall message might be too big for task view
-                    ErrorMessage = string.Join(Environment.NewLine, this.PublishResult.Errors.Where(x => x.MessageType == SqlServer.Dac.DacMessageType.Error || x.MessageType == SqlServer.Dac.DacMessageType.Warning));
-                    throw new Exception(ErrorMessage);
-                }
-            }
-            catch (Exception e)
-            {
-                ErrorMessage = e.Message;
-                Logger.Write(TraceEventType.Error, string.Format("Schema compare publish changes operation {0} failed with exception {1}", this.OperationId, e.Message));
-                throw;
-            }
+        public SchemaComparePublishChangesOperation(SchemaComparisonResult comparisonResult)
+        {
+            Validate.IsNotNull(nameof(comparisonResult), comparisonResult);
+            ComparisonResult = comparisonResult;
         }
+        
+        public abstract void Execute(TaskExecutionMode mode);
 
         // The schema compare public api doesn't currently take a cancellation token so the operation can't be cancelled
```

Comment:
I'm confused by this comment - it looks like you're passing around the cancellation token in other places so is this actually still accurate?

Reply:
Same; removing.